orangecontrib/text/widgets/owcorpusviewer.py

```diff
@@ -460,18 +653,28 @@ def update_splitter(self):
         """
         w1, w2 = self.splitter.sizes()
         ws = w1 + w2
-        if w2 < 2/3 * ws:
-            self.splitter.setSizes([int(ws * 1/3), int(ws * 2/3)])
-
+        if w2 < 2 / 3 * ws:
+            self.splitter.setSizes([int(ws * 1 / 3), int(ws * 2 / 3)])
+
+    @classmethod
+    def migrate_context(cls, context, version):
+        if version < 2:
+            # this setting was changed to match other Orange widgets and to
+            # enable faster documents indexing (when creating output corpora)
+            # I am so sorry but there is no way to transform titles to indices
+            # I will need complete corpus for that
+            context.values.pop("selected_documents", None)
+            # search features was in context by mistake - remove it to allow to
+            # be set correctly by widdget
+            context.values.pop("search_features", None)
```

Comment:
Saved workflows are not opened with correct settings. Maybe you can map `display_indices` into `display_features` and `search_indices` into `search_features` using `display_features`.

I think `display_features` consist of all features.

Reply:
Done. Thank you for the idea.

Reply:
The context is now correctly migrated, but it looks like the last context (not necessarily the right one) is return in case of saved workflow.

Reply:
The problem was that before widget used PerfectContextHandler, which uses a different structure for storing attributes and also stores class_vars separately from attributes. Now I also migrated attributes, metas and class vars.